JustSaying/AwsTools/MessageHandling/MessageDispatcher.cs

```diff
@@ -125,7 +125,9 @@ public async Task DispatchMessage(SQSMessage message, CancellationToken cancella
             {
                 try
                 {
-                    if (!handlingSucceeded && _messageBackoffStrategy != null)
+                    inflightTracker.Complete(message.MessageId);
```

Comment:
I've put this in here before we check for a handler failure to avoid the possible race of updating an individual message visibility timeout (either immediate, or backoff strategy determined) and the batch heartbeat renewal.